lisa/lisa-core/src/main/java/it/unive/lisa/analysis/string/CharInclusion.java

```diff
@@ -0,0 +1,297 @@
+			Collection<Character> maybeContained) {
+		this.certainlyContained = certainlyContained;
+		this.maybeContained = maybeContained;
+	}
+
+	@Override
+	public CharInclusion lubAux(CharInclusion other) throws SemanticException {
+		Collection<Character> lubAuxCertainly = new HashSet<>();
+
+		Collection<Character> lubAuxMaybe = new HashSet<>(this.getMaybeContained());
+		lubAuxMaybe.addAll(other.getMaybeContained());
+
+		for (Character certainlyContainedChar : this.getCertainlyContained())
+			if (other.getCertainlyContained().contains(certainlyContainedChar))
+				lubAuxCertainly.add(certainlyContainedChar);
+
+		return new CharInclusion(lubAuxCertainly, lubAuxMaybe);
+	}
+
+	@Override
+	public CharInclusion wideningAux(CharInclusion other) throws SemanticException {
+		return lubAux(other);
+	}
+
+	@Override
+	public boolean lessOrEqualAux(CharInclusion other) throws SemanticException {
+		if (this.getCertainlyContained().size() > other.getCertainlyContained().size() ||
+				other.getMaybeContained().size() > this.getMaybeContained().size())
+			return false;
+
+		if (!other.getCertainlyContained().containsAll(this.getCertainlyContained()))
+			return false;
+
+		return this.getMaybeContained().containsAll(other.getMaybeContained());
+	}
+
+	@Override
+	public boolean equals(Object o) {
+		if (this == o)
+			return true;
+		if (o == null || getClass() != o.getClass())
+			return false;
+		CharInclusion that = (CharInclusion) o;
+		return Objects.equals(certainlyContained, that.certainlyContained)
+				&& Objects.equals(maybeContained, that.maybeContained);
+	}
+
+	@Override
+	public int hashCode() {
+		return Objects.hash(certainlyContained, maybeContained);
+	}
+
+	@Override
+	public CharInclusion top() {
+		return TOP;
+	}
+
+	@Override
+	public CharInclusion bottom() {
+		return BOTTOM;
+	}
+
+	@Override
+	public boolean isTop() {
+		return this.equals(TOP);
+	}
+
+	@Override
+	public boolean isBottom() {
+		return this.equals(BOTTOM);
+	}
+
+	@Override
+	public DomainRepresentation representation() {
+		if (isBottom())
+			return Lattice.bottomRepresentation();
+		if (isTop())
+			return Lattice.topRepresentation();
+
+		return new StringRepresentation(formatRepresentation());
+	}
+
+	/**
+	 * Yields the set of certainly contained characters of this abstract value.
+	 *
+	 * @return the set of certainly contained characters of this abstract value.
+	 */
+	public Collection<Character> getCertainlyContained() {
+		return this.certainlyContained;
+	}
+
+	/**
+	 * Yields the set of maybe contained characters of this abstract value.
+	 *
+	 * @return the set of maybe contained characters of this abstract value.
+	 */
+	public Collection<Character> getMaybeContained() {
+		return this.maybeContained;
+	}
+
+	private String formatRepresentation() {
+		StringBuilder stringBuilder = new StringBuilder("CertainlyContained: {");
+		int counter = 0;
+
+		for (Character certainlyContainedChar : this.getCertainlyContained()) {
```

Comment:
Instead of looping, you can use `StringUtils.join()` (see [here](https://github.com/lisa-analyzer/lisa/blob/master/lisa/lisa-sdk/src/main/java/it/unive/lisa/analysis/representation/ListRepresentation.java#L66))